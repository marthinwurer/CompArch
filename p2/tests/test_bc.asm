;
; tests BCC/BCS
;
; uses instructions MOV, BPT, ADD, HALT
;
; uses addressing modes 0, 2, 6
;
	.globl	start
;
v1:	.word	0
v2:	.word	177777
v3:	.word	077777
;
start:
	mov	v2,r2
	add	#1,r2
	bcs	ok1
	add	#1,r0
ok1:
	mov	v3,r3
	add	#1,r3
	bcc	ok2
	add	#2,r0
ok2:
;
	bpt
	halt
;
	.end	start
