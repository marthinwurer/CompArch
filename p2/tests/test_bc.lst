      1                                 ;
      2                                 ; tests BCC/BCS
      3                                 ;
      4                                 ; uses instructions MOV, BPT, ADD, HALT
      5                                 ;
      6                                 ; uses addressing modes 0, 2, 6
      7                                 ;
      8 000000                                   .globl  start
      9                                 ;
     10 000000  000000                  v1:      .word   0
     11 000002  177777                  v2:      .word   177777
     12 000004  077777                  v3:      .word   077777
     13                                 ;
     14 000006                          start:
     15 000006  016702  177770                   mov     v2,r2
     16 000012  062702  000001                   add     #1,r2
     17 000016  103402                           bcs     ok1
     18 000020  062700  000001                   add     #1,r0
     19 000024                          ok1:
     20 000024  016703  177754                   mov     v3,r3
     21 000030  062703  000001                   add     #1,r3
     22 000034  103002                           bcc     ok2
     23 000036  062700  000002                   add     #2,r0
     24 000042                          ok2:
     25                                 ;
     26 000042  000003                           bpt
     27 000044  000000                           halt
     28                                 ;
     29 000046  000006'                          .end    start


ok1      000024 R  
ok2      000042 R  
pc     = 000007 %  
r0     = 000000 %  
r1     = 000001 %  
r2     = 000002 %  
r3     = 000003 %  
r4     = 000004 %  
r5     = 000005 %  
r6     = 000006 %  
r7     = 000007 %  
sp     = 000006 %  
start    000006 RG 
v1       000000 R  
v2       000002 R  
v3       000004 R  
