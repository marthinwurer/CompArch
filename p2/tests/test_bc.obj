000 2 00 00
002 2 ff ff
004 2 ff 7f
006 4 c2 1d f8 ff
00a 4 c2 65 01 00
00e 2 02 87
010 4 c0 65 01 00
014 4 c3 1d ec ff
018 4 c3 65 01 00
01c 2 02 86
01e 4 c0 65 02 00
022 2 03 00
024 2 00 00
006
