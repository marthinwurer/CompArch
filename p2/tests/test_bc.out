CPU "ARCH" Simulator, 2.5a(Jan 22 2015)
-----------------------------------------

Memory sets starting address to 6
000006:  PS[00]  016702 177770  MOV 177770(R7),R2
000012:  PS[10]  062702  ADD (R7)+,R2
000016:  PS[05]  103402  BCS 4
000024:  PS[05]  016703 177754  MOV 177754(R7),R3
000030:  PS[01]  062703  ADD (R7)+,R3
000034:  PS[12]  103002  BCC 4
000042:  PS[12]  000003  BPT
  R0[000000]  R1[000000]  R2[000000]  R3[100000]
  R4[000000]  R5[000000]  R6[000000]  R7[000044]
000044:  PS[12]  000000  HALT

Machine Halted - HALT instruction

Simulated time 61 cycles

LAST CPUObject DESTROYED; END OF SIMULATION
